### PR TITLE
layers: Facilitate work on WIP extensions

### PR DESCRIPTION
**Pull request by ShabbyX**
A new empty file (layers/vulkan_wip_ext.h) is added to be automatically
included next to vulkan/vulkan.h in the autogenerated files.
Declarations used by WIP extensions can be placed in this file locally
to support development.

### CODE REVIEW

**Comment by ShabbyX** [comment]
Whether this is the right approach is up for discussion. I was working on validation for a WIP extension and it was not clear how to get the vulkan-header-related changes to VVL, so here's one approach. I recently realized another option would have been to modify my copy of Vulkan-Headers directly.

Let me know if there is any interest in this.

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,6 @@ build-android/third_party
 *.includes
 .vscode/
 .DS_Store
+
+# vim swap files
+.*.sw*
```
Comment (ncesario-lunarg) [inline .gitignore:39]: Been meaning to do this for a while, just keep forgetting :)

Might be worth breaking this out into its own commit, though.

File: layers/vulkan_wip_ext.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Shahbaz Youssefi <syoussefi@google.com>
+ **************************************************************************/
+#pragma once
+
+// This file serves the sole purpose of facilitating implementation of new extensions.  With
+// work-in-program extensions, the relevant part of the Vulkan headers is not available until the
```
Comment (ncesario-lunarg) [inline layers/vulkan_wip_ext.h:20]: "program" -> "progress"?

File: layers/vulkan_wip_ext.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+// extension is merged with the Vulkan registry.  Place that part from a manual generation of the
+// headers in this file which is included in the autogenerated files where necessary.
+//
+// Together with updating registry/vk.xml in Vulkan-Headers and running
+// `./scripts/generate_source.py /path/to/Vulkan-Headers/registry/`, the Mock ICD and DevSim (see
+// docs/creating_tests.md#running-tests-on-devsim-and-mockicd), it's possible to implement
+// validation for new extensions, write unit tests for it and execute them.
+//
+// Do NOT commit those changes to this file.
```
Comment (ncesario-lunarg) [inline layers/vulkan_wip_ext.h:29]: If this file is effectively a "stub," it should probably be included in `.gitignore`. Not sure about other people, but I'll be likely to fall victim to something like `git commit -a ...`.

**Review by ncesario-lunarg — COMMENTED** [review]
I haven't had the pleasure of working on an unpublished extension yet, so I'm probably missing a lot of necessary knowledge here.

If the WIP extension appears in `vk.xml`/Vulkan-Headers then I'm not sure why the `vulkan_wip_ext.h` file is needed. e.g., if the spec changes are on a branch in the Vulkan-Headers repo (not sure if that's possible or not), then I'd think you could change [known_good.json](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/65576d6dc06aca5f2bced8c61f2c95c81add8cbc/scripts/known_good.json#L23) to point to the WIP extension followed by something like
```bash
./scripts/update_deps.py ...
cmake --build /path/to/build-dir --target VulkanVL_generated_source
```
With the first command ensuring VVL picks up the updated vulkan headers and the second making sure the appropriate code has been generated.

Hoping to get some perspectives from other folks.

**Comment by ncesario-lunarg** [comment]
@mark-lunarg just pointed me to https://gitlab.khronos.org/vulkan/Vulkan-ValidationLayers/-/wikis/home#writing-validation-for-new-extensions which describes the "normal" process for working on extensions.

Also, I've been told @mikes-lunarg has a lot of experience with this as well.

**Comment by ShabbyX** [comment]
Thanks for pointing out the gitlab wiki, I didn't know it existed! I'll close this PR.
